Barbarians raid #2588

bob0bob opened the pull request:
I changed how barbarians spawn.  You now have two values in the config file that can alter how barbarians spawn.  
This enhancement only deals with number of bars. spawning.

maxBarbarianSize = 6 - 40
spawnBarbarianSize = 1 -  10
Max will control the max number of bars. that can spawn at one time no matter that the spawn size is.
Spawn size is a value that controls how many bars. will spawn. The game will figure out the normal amount of bars. to spawn and the spawn size is a percentage  1 - 10 which represent 10 - 100%.  So if you only want 50% of the bars. spawning based on what the game wants then put a 5.

This has nothing to go with the difficulty. That is unchanged.
This way you have have hard, with few bars.  or easy two tons of bars.

bob0bob committed:
Barbarians raid #2588

I changed how barbarians spawn.  You now have two values in the config file that can alter how barbarians spawn.  
This enhancement only deals with number of bars. spawning.

maxBarbarianSize = 6 - 40
spawnBarbarianSize = 1 -  10
Max will control the max number of bars. that can spawn at one time no matter that the spawn size is.
Spawn size is a value that controls how many bars. will spawn. The game will figure out the normal amount of bars. to spawn and the spawn size is a percentage  1 - 10 which represent 10 - 100%.  So if you only want 50% of the bars. spawning based on what the game wants then put a 5.

This has nothing to go with the difficulty. That is unchanged.
This way you have have hard, with few bars.  or easy two tons of bars.

Raycoms commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,10 @@
-    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize;
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final double           BARBARIAN_SPAWN_SIZE         = ((double) Configurations.gameplay.spawnBarbarianSize * 0.1);
```
Always complicated to calc statics like that, if the player changes the config from the ingame editor it wouldn't reload this variable then.

Raycoms replied:
I know the code was like that before, but ti would be nice to use the Configurations at the places directly, they are constants already basically.

bob0bob replied:
Done

bob0bob committed:
Changing it to a new style for config file.

raycoms said would likea direct use of config variable so person doesn't have to reload after altering the variable in the config file.

bob0bob commented:
Ok I made the change to use the variable directly.

Raycoms commented:
:shipit: 

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2692/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
     public static final ResourceLocation ARCHER                       = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
     public static final ResourceLocation ARCHER                       = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
     public static final ResourceLocation CHIEF                        = new ResourceLocation(Constants.MOD_ID, "ChiefBarbarian");
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
-    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize;
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              BARBARIAN_MAX_SIZE           = Configurations.gameplay.maxBarbarianSize;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              BARBARIAN_MAX_SIZE           = Configurations.gameplay.maxBarbarianSize;
+    public static final double           BARBARIANS_MULTIPLIER        = 0.6;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              BARBARIAN_MAX_SIZE           = Configurations.gameplay.maxBarbarianSize;
+    public static final double           BARBARIANS_MULTIPLIER        = 0.6;
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.30;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
+    public static final int              BARBARIAN_MAX_SIZE           = Configurations.gameplay.maxBarbarianSize;
+    public static final double           BARBARIANS_MULTIPLIER        = 0.6;
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.30;
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
+    public static final double           BARBARIANS_MULTIPLIER        = 0.6;
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.30;
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
     public static final int              PREFERRED_MAX_HORDE_SIZE     = 40;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.30;
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
     public static final int              PREFERRED_MAX_HORDE_SIZE     = 40;
     public static final int              PREFERRED_MAX_BARBARIANS     = 22;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 38233 |  1  |  5  |  -4  |
| Errors |  592  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=13763&tab=Inspection&buildTypeId=bt88)

Raycoms commented:
There are some conflicts with your other pr.

bob0bob committed:
Formatting changes.

conflict between 2 different PR altering the same files.

bob0bob commented:
Is there anything this one needs?   I appears to be all corrected?  let me know.

Raycoms committed:
Merge branch 'version/1.12' into Barbarians-raid-#2588
